backend/app/services/db/listing_manager.py

```diff
@@ -109,6 +118,10 @@ def search_listings(username: str = None, title_keywords: str = None,
                 values.append(user_details['user_id'])
                 conditions.append("user_id = %s")
 
+        if rating is not None:
+            values.append(rating)
+            conditions.append("avg_user_rating >= %s")

```

Comment:
How about you create another table average_ratings, store average rating and number of ratings, and update it for all rating changes?